cmd/copy.go

```diff
@@ -876,13 +899,67 @@ func (cca *cookedCopyCmdArgs) processCopyJobPartOrders() (err error) {
 	// depending on the source and destination type, we process the cp command differently
 	// Create enumerator and do enumerating
 	switch cca.fromTo {
-	case common.EFromTo.LocalBlob():
-		fallthrough
-	case common.EFromTo.LocalBlobFS():
-		fallthrough
-	case common.EFromTo.LocalFile():
-		e := copyUploadEnumerator(jobPartOrder)
-		err = e.enumerate(cca)
+	case common.EFromTo.LocalBlob(),
+		common.EFromTo.LocalBlobFS(),
+		common.EFromTo.LocalFile():
+		var traverser resourceTraverser
+		var destTraverser resourceTraverser
+
+		dst := cca.destination
+
+		if cca.destinationSAS != "" {
+			destURL, err := url.Parse(dst)
+
+			if err != nil {
+				return err
+			}
+
+			destURL = copyHandlerUtil{}.appendQueryParamToUrl(destURL, cca.destinationSAS)
+			dst = destURL.String()
+		}
+
+		traverser, err = initResourceTraverser(cca.source, cca.fromTo.From(), nil, nil, &cca.followSymlinks, &cca.listOfFilesLocation, cca.recursive, func() {})
+		if err != nil {
+			return err
+		}
+
+		destTraverser, err = initResourceTraverser(dst, cca.fromTo.To(), &ctx, &cca.credentialInfo, &cca.followSymlinks, nil, cca.recursive, func() {})
+		if err != nil {
+			return err
+		}
+
+		isDir := traverser.isDirectory(false)
+		isDestDir := destTraverser.isDirectory(true)
+
+		if isDir && !cca.recursive {
+			return errors.New("cannot copy from container or directory without --recursive or trailing wildcard (/*)")
+		}
+
+		filters := cca.initModularFilters()
+		processor := func(object storedObject) error {
+			src := cca.findObject(true, isDestDir, object)
+			// Why hand in the source on the destination? Because we need to adjust the path for the source if there's no wildcard.
```

Comment:
I struggeld to understand what this comment is trying to say, or which of the two lines it refers to.  I think "object" = "sourceObject" and this is referring to the dst line... but it took me a while to figure that out and I'm not 100% sure I've got it right.  Maybe whitespace before the comment would help, and maybe rename object to sourceObject?

Reply:
OK so this was a comment before cca.findObject had changed its signature and I need to remove it.

Initially, cca.findObject took in a path. I handed in the source path there.

Reply:
Yes, already is removed.